src/lib/dnssd/minimal_mdns/Parser.cpp

```diff
@@ -66,23 +66,24 @@ bool QueryData::Parse(const BytesRange & validData, const uint8_t ** start)
     return true;
 }
 
-bool QueryData::Append(HeaderRef & hdr, chip::Encoding::BigEndian::BufferWriter & out) const
+bool QueryData::Append(HeaderRef & hdr, RecordWriter & out) const
 {
     if ((hdr.GetAdditionalCount() != 0) || (hdr.GetAnswerCount() != 0) || (hdr.GetAuthorityCount() != 0))
     {
         return false;
     }
 
-    GetName().Put(out);
-    out.Put16(static_cast<uint16_t>(mType));
-    out.Put16(static_cast<uint16_t>(mClass) | (mAnswerViaUnicast ? kQClassUnicastAnswerFlag : 0));
+    out.WriteQName(GetName());
+    out.Writer().Put16(static_cast<uint16_t>(mType));
```

Comment:
nit: Suggest repacing `out.Writer()` with a helper variable `auto & outWriter = out.Writer()` and use that everywhere.

Reply:
I added the Put/Fit methods into the RecordWriter, this should make the code easier to read and follow.